### PR TITLE
Show products of sub category when main category is clicked in menu

### PR DESCRIPTION
**Pull request by harm-meijer**
Show products of sub category when main category is clicked in meny
Search text box

## Checklist
* [x] Unit tests
* [x] E2E tests
* [x] Documentation

### CODE REVIEW

**Comment by harm-meijer** [comment]
> When searching in all categories the breadcrumb disappears as it can't find category 'all'

I will fix that in the next pull, link such as http://localhost:8080/products/non-existing-category will assume "all" as category but should produce "Category not found"

**Comment by plaskevich** [comment]
When viewing product list, images are squished a little in height

File: src/components/productoverview/ProductList.vue [hunk 1]
```diff
@@ -47,6 +50,27 @@
...
+          class="scroll-to-top"
+          v-scroll-to="{
+            el: '#form-filter-products',
+            duration: 500,
+            easing: 'linear',
+            offset: -200,
+          }"
+          v-vpshow="showScroll"
+        >
+            <span class="scroll-to-top-text"> go to top </span>
```
Comment (plaskevich) [inline src/components/productoverview/ProductList.vue]: I think this should be an i18n, for both german and english

**Comment by harm-meijer** [comment]
Manually merged due to conflicts with rebase merge

**Pull request closed without merging**